app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -0,0 +1,619 @@
+package cm.aptoide.pt.app.view;
+
+import android.text.TextUtils;
+import android.text.format.DateUtils;
+import cm.aptoide.accountmanager.AptoideAccountManager;
+import cm.aptoide.pt.R;
+import cm.aptoide.pt.account.AccountAnalytics;
+import cm.aptoide.pt.account.view.AccountNavigator;
+import cm.aptoide.pt.actions.PermissionManager;
+import cm.aptoide.pt.actions.PermissionService;
+import cm.aptoide.pt.app.AppViewAnalytics;
+import cm.aptoide.pt.app.AppViewManager;
+import cm.aptoide.pt.app.DownloadAppViewModel;
+import cm.aptoide.pt.crashreports.CrashReport;
+import cm.aptoide.pt.logger.Logger;
+import cm.aptoide.pt.presenter.Presenter;
+import cm.aptoide.pt.presenter.View;
+import cm.aptoide.pt.share.ShareDialogs;
+import java.util.concurrent.TimeUnit;
+import rx.Completable;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Single;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.exceptions.OnErrorNotImplementedException;
+
+/**
+ * Created by franciscocalado on 08/05/18.
+ */
+
+public class AppViewPresenter implements Presenter {
+  private static final long TIME_BETWEEN_SCROLL = 2 * DateUtils.SECOND_IN_MILLIS;
```

Comment:
We should not use DateUtils just for this, it makes us depend on it and might not facilitate testing. Can replace for 1000